pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -1054,6 +1067,29 @@ def check_genesis_state(self):
             self.kill_neard()
             self.make_initial_backup()
 
+    def run_restore_from_backup_cmd(self, backup_path):
+        logging.info(f'restoring data dir from backup at {backup_path}')
+        # Before using snapshots for backup we used to copy the 'data' folder.
+        # This is useful to be able to restore from backups done the old way.
+        if not os.path.exists(os.path.join(backup_path, 'data')):
+            shutil.copytree(backup_path, self.target_near_home_path('data'))
+            exit_code = 0
+        else:
+            cmd = [
+                self.data['current_neard_path'], '--home',
+                backup_path,
+                '--unsafe-fast-startup',
+                'database',
+                'make-snapshot',
+                '--destination',
+                self.target_near_home_path()
+            ]
+            self.run_neard(cmd)
+            running = True
+            while running:
```

Comment:
same about this `while running` thing being problematic as I mentioned above. you can just use `subprocess.check_call()`